File: solutions/c/leap/1/leap.c
#include "leap.h"
#include <stdbool.h>

bool leap_year(int year) {
    // A year is a leap year if it is divisible by 4
    // AND (not divisible by 100 OR divisible by 400)
    if (year % 4 == 0) {
        if (year % 100 !=0 || year % 400 == 0){
            return true;
        }
    }
    return false;
}

File: solutions/java/leap/1/src/main/java/Leap.java
class Leap {

    boolean isLeapYear(int year) {
        /**
        * Every year divisible by 4...
        * EXCEPT years divisible by 100...
        * UNLESS they are also divisible by 400.
        */
        return (year % 4 == 0) && (year % 100 != 0 || year % 400 == 0);
    }

}
